internal/wrr/edf.go

```diff
@@ -24,22 +24,22 @@ import (
 
 // edfWrr is a struct for EDF weighted round robin implementation.
 type edfWrr struct {
-	lock  sync.Mutex
-	items edfPriorityQueue
+	lock    sync.Mutex
+	items   edfPriorityQueue
+	weights map[interface{}]int64
```

Comment:
What if this is a `map[interface{}]*edfEntry`, and `edfEntry` keeps the item index (that gets updated in `Swap`)?

The item index should help with `Update` and `Remove` later.